cicd-scripts/app_install.sh

```diff
@@ -1,51 +1,68 @@
 #!/bin/bash
 
+
+# A hack to get the environment running without ansible local env variables
+# This block of code will eventually be removed once ansible task is completed
+SPIDER_PYTHON_VERSION=3.12
+SPIDER_STAGING_URLS_API=https://staging.search.usa.gov/urls
+spider_local_path=/etc/profile.d/spider_local.sh
+
+# Writing environment variables to the profile file
+echo "
+export SPIDER_PYTHON_VERSION=${SPIDER_PYTHON_VERSION}
+export SPIDER_STAGING_URLS_API=${SPIDER_STAGING_URLS_API}
+" | tee "$spider_local_path" > /dev/null
+
+# Source the script to update the current shell's environment
+source "$spider_local_path"
+### TODO: Remove the above code block after ansible is fully implmented
+
+
 # Update and upgrade the system without prompting for confirmation
 sudo apt-get update -y
 sudo apt-get upgrade -y
 
 # Install necessary system dependencies
 sudo apt-get install -y python-setuptools python-pip
 
-# Function to install Python 3.12
 install_python() {
-    echo "Installing Python 3.12..."
+    echo "Installing ${SPIDER_PYTHON_VERSION}"
     sudo apt-get install -y build-essential checkinstall libreadline-dev \
                             libncursesw5-dev libssl-dev libsqlite3-dev \
                             tk-dev libgdbm-dev libc6-dev libbz2-dev \
                             zlib1g-dev openssl libffi-dev
 
-    # Download Python 3.12 source code
+    # Download Python source code
     cd /usr/src
-    sudo wget https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tgz
-    sudo tar xzf Python-3.12.0.tgz
+    sudo wget https://www.python.org/ftp/python/${SPIDER_PYTHON_VERSION}.0/Python-${SPIDER_PYTHON_VERSION}.0.tgz
+    sudo tar xzf Python-${SPIDER_PYTHON_VERSION}.0.tgz
```

Comment:
does this mean we still have to make changes here for minor version updates?  Why not include the ".0" in the variable and the just use `python3` or `python` instead of `python 3.12` when upgrading pip below.

Reply:
This is based on semantic versioning, and some places in the code require only the major and a minor, eg: `3.12` If my variable also has the patch suffix `.0` then I would have to remove it where ever it's incompatible. So, basically it's easier to add the patch version rather than removing it. I didn't give it much thought though, because usually patch only has minor fixes/improvements and no breaking changes